### ISSUE TITLE
Documentation

### ISSUE THREAD

**Issue by evilsoft** [evilsoft/crocks#42]
## All the Things

Would like to have these bits get documented with the following general organization in mind:
* Each `crock`  gets its own API file
  * Signature
  * Sales Pitch for using the specific `crock`
  * Table of functions implemented on the `crock` (By Constructor and Instance).
  * Section for Constructor Functions
  * Section for Instance Functions
* One API file for all `Monoids`,
  * High level description of what a Monoid is and how to use them.
  * Examples showing General use cases (pick a couple)
  * Table of common functions shared between them
  * Section for Each Monoid and its API.
* One API file for each of the function classifications.

- [ ] crocks
  - [ ] `Arrow`
  - [ ] `Async`
  - [ ] `Const`
  - [ ] `Either`
  - [ ] `Free` _(still to come)_
  - [ ] `Identity`
  - [ ] `IO`
  - [ ] `List`
  - [ ] `Maybe`
  - [ ] `Pred`
  - [ ] `Reader`
  - [ ] `Star`
  - [ ] `State`
  - [ ] `Unit`
  - [ ] `Validation` _(still to come)_
  - [ ] `Writer`

- [ ] Monoids
 
- [ ] Functions
  - [ ] Combinator functions
  - [ ] Helper functions
  - [ ] Logic functions _(still to come)_
  - [ ] Predicate functions
  - [ ] Pointfree functions

**Comment by rstegg** [comment]
> Sales Pitch for using the specific crock

should be 

> TL;DR of what the specific crock does

**Comment by dalefrancis88** [comment]
@evilsoft are there anyof these I can take off your hands?

**Comment by StevenLangbroek** [comment]
Hey all! I'm starting to work crocks into some of our infrastructure, and think you could make a good case for having some guides or tutorials, particularly about how to combine crocks in common use-cases (e.g., I had to combine state and async and had some trouble figuring out a good / clean approach). Would you be interested in contributions in that area?

**Comment by dalefrancis88** [comment]
Hey man, welcome to the world of `crocks` glad you're here with us 👍 

We're always happy when people want to contribute and help, especially around documentation and guides. We actually have an issue for it here, https://github.com/evilsoft/crocks/issues/196 

If you need any help getting things set up or what to put where just reach out, happy to help where we can.

**Comment by StevenLangbroek** [comment]
Thanks @dalefrancis88, working with crocks has been a pleasure so far, and if the welcome is any indicator, I look forward to the community just as much :).

I totally missed that issue! 🙈 

So, the type annotation we can maybe lift from [`fantasyland`](https://github.com/fantasyland/fantasy-land#type-signature-notation) with an attribution maybe?

I'm still building up general knowledge around category theory et al, so maybe my efforts would be best spent on beginner'ish guides?

**Comment by dalefrancis88** [comment]
Your best efforts are wherever you want to put them. I am doing what I can to learn by going helping and chatting on the Gitter channel and helping to write the remaining docs. I can say for sure that from my experience, writing the documentation has helped me greatly in both understanding `crocks` and understanding category theory.
